app/screens/send-bitcoin-screen/send-lnurl-screen.tsx

```diff
@@ -0,0 +1,323 @@
+        {value}
+      </Text>
+    </View>
+  )
+}
+
+export const SendLNUrlScreen: React.FC<Props> = ({ route }: Props) => {
+  const { navigate } = useNavigation()
+  const client = useApolloClient()
+  const balance = balanceBtc(client)
+
+  const btcPrice = useBTCPrice()
+  const { primaryCurrency, secondaryCurrency, toggleCurrency } = useCurrencies()
+  const [primaryAmount, convertPrimaryAmount, setPrimaryAmount, setPrimaryAmountValue] =
+    useMoneyAmount(secondaryCurrency)
+
+  const [
+    secondaryAmount,
+    convertSecondaryAmount,
+    setSecondaryAmount,
+    setSecondaryAmountValue,
+  ] = useMoneyAmount(secondaryCurrency)
+
+  const [amount, setAmount] = useState(0)
+  const [status, setStatus] = useState(PAYMENT_STATUS.Pending)
+  const [submitStatus, setSubmitStatus] = useState(0)
+  const [callback, setCallback] = useState("")
+  const [minSendable, setMinSendable] = useState(null)
+  const [maxSendable, setMaxSendable] = useState(null)
+  const [description, setDescription] = useState("")
+  const [memo, setMemo] = useState("")
+  const [maxMemoSize, setMaxMemoSize] = useState(0)
+
+  const [queryTransactions] = useLazyQuery(QUERY_TRANSACTIONS, {
+    fetchPolicy: "network-only",
+  })
+
+  const [payInvoice] = useMutation(PAY_INVOICE, {
+    update: () => queryTransactions(),
+  })
+
+  const hasEnoughBalance = balance >= amount
+  const isAmountSendable = isAmountValid(amount, minSendable, maxSendable)
+
+  useEffect(() => {
+    const parseInitialInvoice = async () => {
+      const lnurl = await parseUrl(route.params.invoice)
+      const { tag, callback, minSendable, maxSendable, commentAllowed, metadata } = lnurl
+
+      if (!isProtocolSupported(tag)) {
+        Alert.alert("LNUrl protocol not supported")
+        throw new Error("LNURL protocol not supported")
+      }
+
+      setAmount(mSatToSat(minSendable))
+      setPrimaryAmountValue(mSatToSat(minSendable))
+      setSecondaryAmountValue(mSatToSat(minSendable))
+      setCallback(callback)
+      setMinSendable(mSatToSat(minSendable))
+      setMaxSendable(mSatToSat(maxSendable))
+      setMaxMemoSize(commentAllowed)
+
+      try {
+        const description = JSON.parse(metadata)[0][1]
+        setDescription(description)
+      } catch (error) {
+        setDescription("Invoice description not recognized")
+      }
+    }
+
+    parseInitialInvoice()
+  }, [route.params.invoice, setPrimaryAmountValue, setSecondaryAmountValue])
+
+  const submitPayment = async () => {
+    setSubmitStatus(1)
+
+    try {
+      const paymentInvoice = await invoiceRequest(callback, amount, memo)
+      const payment = await payInvoice({
+        variables: { invoice: paymentInvoice, memo: description },
+      })
+      const { data } = payment
+
+      setSubmitStatus(0)
+
+      if (
+        data.invoice.payInvoice === "success" ||
+        data.invoice.payInvoice === "pending"
+      ) {
+        setStatus(PAYMENT_STATUS.Success)
```

Comment:
Need to call [queryWallet](https://github.com/GaloyMoney/galoy-mobile/blob/009a5b89caa4f63989b062bef0a05c1fb11badb3/app/screens/send-bitcoin-screen/send-bitcoin-confirmation-screen.tsx#L355) so that balance is updated.